### DISCUSSION TITLE
Tree with rounded corners?

### DISCUSSION THREAD

**Discussion by Bebby6** [spectreconsole/spectre.console#1121] [General]
![image](https://user-images.githubusercontent.com/121303220/209443416-f77de5a1-a368-4a06-aa2f-7ea69c4800ee.png)

Is there a way to have a rounded corners, like Table?
![image](https://user-images.githubusercontent.com/121303220/209443493-e186b656-13c0-4599-84b5-707cdcee9354.png)

Thanks. Probably I can make a new Guide for it, but I don't know how.

**Comment by 0xced** [comment]
Here's a guide using `╰` (U+2570 / BOX DRAWINGS LIGHT ARC UP AND RIGHT) instead of `└` (U+2514 / BOX DRAWINGS LIGHT UP AND RIGHT) that will render a rounded corner.

```csharp
public class RoundedLineTreeGuide : TreeGuide
{
    public override string GetPart(TreeGuidePart part)
    {
        return part switch
        {
            TreeGuidePart.Space => "    ",
            TreeGuidePart.Continue => "│   ",
            TreeGuidePart.Fork => "├── ",
            TreeGuidePart.End => "╰── ",
            _ => throw new ArgumentOutOfRangeException(nameof(part), part, "Unknown tree part."),
        };
    }
}
```

Here's what it looks like when used in the `Trees` example code.
<img width="422" alt="Screenshot 2022-12-26 at 12 24 28" src="https://user-images.githubusercontent.com/51363/209543700-00cd5cd6-258d-4141-b05c-cb134142da5d.png">
